pkg/provider/azure_instances.go

```diff
@@ -231,6 +231,10 @@ func (az *Cloud) InstanceExists(ctx context.Context, node *v1.Node) (bool, error
 		var err error
 		providerID, err = cloudprovider.GetInstanceProviderID(ctx, az, types.NodeName(node.Name))
 		if err != nil {
+			if strings.Contains(err.Error(), cloudprovider.InstanceNotFound.Error()) {
```

Comment:
Do you think we can use `errors.Is()`? Like here:
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/c6f9190091473cabf094f89de2bb6d93dba20114/pkg/provider/azure_instances.go#L206

Reply:
Please ignore. It is the same style like L301